presto-main/src/test/java/com/facebook/presto/cost/TestCostComparator.java

```diff
@@ -43,6 +43,15 @@ public void testMemoryWeight()
                 .assertCompare();
     }
 
+    @Test
+    public void testMemoryLimit()
+    {
+        new CostComparisonAssertion(1.0, 0.0, 0.0)
+                .smaller(1000, 100, 100)
+                .larger(100, 6e13, 100)
+                .assertCompare();
```

Comment:
can you also add a test where you compare the case where individual memory costs are higher than query max memory